src/main/java/org/jenkinsci/plugins/azurekeyvaultplugin/AzureKeyVaultGlobalConfiguration.java

```diff
@@ -163,6 +167,30 @@ private Optional<String> getPropertyByEnvOrSystemProperty(String envVariable, St
         return Optional.empty();
     }
 
+    private Optional<String> getPropertyByFile(String envVariable, String systemProperty) {
+        String envResult = System.getenv(envVariable);
+        if (envResult != null) {
+            Path pathToSecret = Paths.get(envResult);
+            try {
+                return Optional.of(Files.readAllLines(pathToSecret).get(0));
+            } catch (IOException e) {
+                throw new RuntimeException(e);
```

Comment:
more appropriate exception
```suggestion
                throw new UncheckedIOException(e);
```